src/components/Maps/SearchBar.tsx

```diff
@@ -86,9 +89,24 @@ const SearchBar = ({
               InputProps={{
                 endAdornment: (
                   <InputAdornment position="end">
-                    <IconButton>
-                      <SearchIcon />
-                    </IconButton>
+                    {address && (
+                      <>
+                        <IconButton
+                          onClick={() => {
+                            setAddress('');
+                          }}
+                        >
+                          <ClearIcon />
+                        </IconButton>
+                      </>
+                    )}
+                    {!address && (
+                      <>
+                        <IconButton>
+                          <SearchIcon />
+                        </IconButton>
+                      </>
+                    )}
```

Comment:
nit: can't this just go in a ternary? also you don't need the `<>` tags if you only have one node within them - i think they're called fragments and they're meant to be used if you have multiple sibling tags you're dealing with (and i believe jsx only ever lets you return one normally)